File: Samples/JotFormTest/Classes/Controller/RegisterUserViewController.m
//
//  RegisterUserViewController.m
//  JotFormTest
//
//  Created by Administrator on 1/8/14.
//  Copyright (c) 2014 wang. All rights reserved.
//

#import "RegisterUserViewController.h"
#import "SVProgressHUD.h"

@interface RegisterUserViewController ()

@property (nonatomic,weak) IBOutlet UITextField *usernameTextField;
@property (nonatomic,weak) IBOutlet UITextField *passwordTextField;
@property (nonatomic,weak) IBOutlet UITextField *emailTextField;

@end

@implementation RegisterUserViewController

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    self.title = @"Register user";
}

#pragma mark - User definition method

- (void)registerUser {
    if (self.usernameTextField.text.length == 0) {
        [self.usernameTextField becomeFirstResponder];
        return;
    }
    
    if (self.passwordTextField.text.length == 0) {
        [self.passwordTextField becomeFirstResponder];
        return;
    }
    
    if (self.emailTextField.text.length == 0) {
        [self.emailTextField becomeFirstResponder];
        return;
    }
    
    [SVProgressHUD showWithStatus:@"Registering user..."];
    
    NSMutableDictionary *userInfo = [[NSMutableDictionary alloc] init];
    userInfo[@"username"] = self.usernameTextField.text;
    userInfo[@"password"] = self.passwordTextField.text;
    userInfo[@"email"] = self.emailTextField.text;
    
    /* [[SharedData sharedData].apiClient registerUser:userInfo onSuccess:^(id result) {
        [SVProgressHUD dismiss];
        
        if (result) {
            NSInteger responseCode = [result[@"responseCode"] integerValue];
            
            if (responseCode == 200 || responseCode == 206) {
                UIAlertController *alertView = [UIAlertController
                                                alertControllerWithTitle:@"JotFormAPISample"
                                                message:@"You registered new user successfully."
                                                preferredStyle:UIAlertControllerStyleAlert];
                
                UIAlertAction *cancelButton = [UIAlertAction
                                               actionWithTitle:@"Ok"
                                               style:UIAlertActionStyleDefault
                                               handler:^(UIAlertAction *action){
                                                   
                                               }];
                
                [alertView addAction:cancelButton];
                [self presentViewController:alertView animated:YES completion:nil];
            } else {
                UIAlertController *alertView = [UIAlertController
                                                alertControllerWithTitle:@"JotFormAPISample"
                                                message:result[@"message"]
                                                preferredStyle:UIAlertControllerStyleAlert];
                
                UIAlertAction *cancelButton = [UIAlertAction
                                               actionWithTitle:@"Ok"
                                               style:UIAlertActionStyleDefault
                                               handler:^(UIAlertAction *action){
                                                   
                                               }];
                
                [alertView addAction:cancelButton];
                [self presentViewController:alertView animated:YES completion:nil];
            }
        }
    } onFailure:^(id error) {
        [SVProgressHUD dismiss];
    }]; */
}

#pragma mark - IBAction

- (IBAction)registerButtonClicked:(id)sender {
    [self registerUser];
}


@end

File: Samples/JotFormTest/Classes/Controller/RegisterUserViewController.swift
//
//  RegisterUserViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 2/4/18.
//  Copyright © 2018 wang. All rights reserved.
//

import Foundation

class RegisterUserViewController: UIViewController {
    @IBOutlet weak var usernameTextField: UITextField!
    @IBOutlet weak var passwordTextField: UITextField!
    @IBOutlet weak var emailTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        title = "Register user"
    }
    
    func registerUser() {
        if  usernameTextField.text?.count == 0 {
            usernameTextField.becomeFirstResponder()
            return
        }
       
        if passwordTextField.text?.count == 0 {
            passwordTextField.becomeFirstResponder()
            return
        }
        
        if emailTextField.text?.count == 0 {
           emailTextField.becomeFirstResponder()
           return
        }
        
        SVProgressHUD.show(withStatus: "Registering user...")
        
        var userInfo = [String: Any]()
        userInfo["username"] = usernameTextField.text
        userInfo["password"] = passwordTextField.text
        userInfo["email"] = emailTextField.text
        
        SharedData.sharedData.apiClient?.registerUser(userInfo, onSuccess: {(_ result: Any) -> Void in
            SVProgressHUD.dismiss()
            
            let alertView = UIAlertController(title: "JotFormAPISample", message: "You registered new user successfully.", preferredStyle: .alert)
            let cancelButton = UIAlertAction(title: "Ok", style: .default, handler: {(_ action: UIAlertAction) -> Void in
            })
            alertView.addAction(cancelButton)
            self.present(alertView, animated: true, completion: nil)
            
        }, onFailure: {(_ error: Any) -> Void in
            SVProgressHUD.dismiss()
        })
    }
    
    @IBAction func registerButtonClicked(_ sender: Any) {
        registerUser()
    }
}
